packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectHelper.kt

```diff
@@ -221,6 +221,13 @@ internal object RealmObjectHelper {
         value: R
     ) {
         val o = obj.`$realm$ObjectPointer` ?: throw IllegalStateException("Invalid/deleted object")
+        obj.`$realm$metadata`!!.let { classMetaData ->
```

Comment:
Unfortunately, we cannot prevent updates altogether since a migration might move the primary key column to a new one, which will have all default values which must be set to valid values.

We work around that in Java since people can populate a column before marking it the new primary key